zerver/data_import/hipchat.py

```diff
@@ -0,0 +1,236 @@
+    flat_data = [
+        d['User']
+        for d in raw_data
+    ]
+
+    flat_data = [
+        d
+        for d in flat_data
+        if not d['is_deleted']
+    ]
+
+    def is_realm_admin(v: str) -> bool:
+        if v == 'user':
+            return False
+        elif v == 'admin':
+            return True
+        else:
+            raise Exception('unexpected value')
+
+    '''
+    TODO:
+        avatar
+        timezone
+    '''
+
+    def process(in_dict: ZerverFieldsT) -> ZerverFieldsT:
+        out_dict = dict()
+
+        out_dict['is_realm_admin'] = is_realm_admin(in_dict['account_type'])
+        out_dict['email'] = in_dict['email']
+        out_dict['delivery_email'] = in_dict['email']
+        out_dict['id'] = in_dict['id']
+        out_dict['short_name'] = in_dict['mention_name']
+        out_dict['full_name'] = in_dict['name']
+
+        # unmapped fields:
+        #    title - Developer, Project Manager, etc.
+        #    rooms - no good sample data
+        #    created - we just use "now"
+        #    roles - we just use account_type
+        return out_dict
+
+    return list(map(process, flat_data))
+
+def user_profile_dict(in_dict: ZerverFieldsT, realm_id: int) -> ZerverFieldsT:
+    now = int(timezone_now().timestamp())
+    obj = UserProfile(
+        realm_id=realm_id,
+        date_joined=now,
+        pointer=-1,
+        avatar_source='U',
+        **in_dict
+    )
+    dct = model_to_dict(obj)
+    return dct
+
+def read_room_data(data_dir: str) -> List[ZerverFieldsT]:
+    fn = 'rooms.json'
+    data_file = os.path.join(data_dir, fn)
+    data = json.load(open(data_file))
+    return data
+
+def clean_room_data(raw_data: List[ZerverFieldsT]) -> List[ZerverFieldsT]:
+    flat_data = [
+        d['Room']
+        for d in raw_data
+    ]
+
+    flat_data = [
+        d
+        for d in flat_data
+        if not d['is_archived']
+    ]
+
+    def invite_only(v: str) -> bool:
+        if v == 'public':
+            return False
+        elif v == 'private':
+            return True
+        else:
+            raise Exception('unexpected value')
+
+    def process(in_dict: ZerverFieldsT) -> ZerverFieldsT:
+        out_dict = dict()
+
+        out_dict['id'] = in_dict['id']
+        out_dict['name'] = in_dict['name']
+        out_dict['description'] = in_dict['topic']
+        out_dict['invite_only'] = invite_only(in_dict['privacy'])
+
+        # unmapped fields:
+        #    guest_access_url: no Zulip equivalent
+        #    created: we just use "now"
+        #    members: no good sample data
+        #    owners: no good sample data
+        #    participants: no good sample data
+        return out_dict
+
+    return list(map(process, flat_data))
+
+def stream_dict(in_dict: ZerverFieldsT, realm_id: int) -> ZerverFieldsT:
+    now = int(timezone_now().timestamp())
+    obj = Stream(
+        realm_id=realm_id,
+        date_created=now,
+        **in_dict
+    )
+    dct = model_to_dict(obj)
+    return dct
```

Comment:
Same story as with `UserProfile`; this should probably be calling `build_stream` inside `process` above.  

(I want to use the `build_stream` type helpers from `import_util` in part for code deduplication and in part because we plan to add additional sanity-check validation to those functions that would be annoying to have in each export tool).